### PR TITLE
GroundPrimitive batching

### PR DESCRIPTION
**Pull request by bagnell**
Batches `GroundPrimitive`s as long as they are the same color.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Change Log
 
 ### 1.18 - 2016-02-01
 
+* Deprecated
+    * Deprecated `GroundPrimitive.geometryInstance`. It will be removed in 1.20. Use `GroundPrimitive.geometryInstances` instead.
```
Comment (pjcozzi) [inline CHANGES.md]: Submit an issue for this.

File: Source/Scene/GroundPrimitive.js [hunk 2]
```diff
@@ -694,37 +725,80 @@ define([
...
+                    instance = instances[i];
+                    geometry = instance.geometry;
+
+                    instanceType = geometry.constructor;
+                    if (defined(instanceType) && defined(instanceType.createShadowVolume)) {
+                        var attributes = instance.attributes;
+
+                        //>>includeStart('debug', pragmas.debug);
+                        if (!defined(attributes) || !defined(attributes.color)) {
+                            throw new DeveloperError('Not all of the geometry instances have the same color attribute.');
```
Comment (pjcozzi) [inline Source/Scene/GroundPrimitive.js]: Document this exception in the `update` reference doc.

File: Source/Scene/GroundPrimitive.js [hunk 3]
```diff
@@ -111,7 +115,7 @@ define([
      * scene.primitives.add(new Cesium.GroundPrimitive({
      *   geometryInstance : rectangleInstance
```
Comment (pjcozzi) [inline Source/Scene/GroundPrimitive.js]: Update the sample to not use the deprecated parameter.  Perhaps also add a second example showing batch with a comment saying that the geometries are the same color.

File: Source/Scene/GroundPrimitive.js [hunk 4]
```diff
@@ -694,37 +725,80 @@ define([
...
+                            throw new DeveloperError('Not all of the geometry instances have the same color attribute.');
+                        } else if (!defined(color)) {
+                            color = attributes.color;
+                        }
+                        //>>includeEnd('debug');
+
+                        groundInstances[i] = new GeometryInstance({
+                            geometry : instanceType.createShadowVolume(geometry, computeMinimumHeight, computeMaximumHeight),
+                            attributes : attributes,
+                            modelMatrix : Matrix4.IDENTITY,
```
Comment (pjcozzi) [inline Source/Scene/GroundPrimitive.js]: Is `Matrix4.IDENTITY` the default?

File: CHANGES.md (hunk 1 above)
Comment (pjcozzi) [inline CHANGES.md]: Also mention that ground primitive now supports batching for better performance.

**Comment by pjcozzi** [comment]
Update the Sandcastle example to include a set of batched geometries with the ground primitive.

**Comment by bagnell** [comment]
@pjcozzi This is ready.

**Comment by pjcozzi** [comment]
Let me know when this is ready with the latest picking fixes we discussed offline.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another review.

**Comment by pjcozzi** [comment]
Looks awesome!

**Pull request merged**